lib/chef/provider_resolver.rb

```diff
@@ -68,27 +68,39 @@ def provided_by?(provider_class)
       potential_handlers.include?(provider_class)
     end
 
-    def self.includes_handler?(resource_name, provider_class)
-      priority_map.includes_handler?(resource_name, provider_class)
-    end
-
     def enabled_handlers
-      @enabled_handlers ||=
-        potential_handlers.select { |handler| !overrode_provides?(handler) || handler.provides?(node, resource) }
+      potential_handlers.select { |handler| !overrode_provides?(handler) || handler.provides?(node, resource) }
```

Comment:
Nope. I don't think we need to do it as many places as we are, though.

On Wed, Jul 1, 2015, 4:53 PM Lamont Granquist <notifications@github.com>
wrote:

> In lib/chef/provider_resolver.rb
> <https://github.com/chef/chef/pull/3627#discussion_r33733512>:
>
> >      def enabled_handlers
> > -      @enabled_handlers ||=
> > -        potential_handlers.select { |handler| !overrode_provides?(handler) || handler.provides?(node, resource) }
> > +      potential_handlers.select { |handler| !overrode_provides?(handler) || handler.provides?(node, resource) }
>
> yeah, do we have to drop the memoization here, and below, and why?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/chef/chef/pull/3627/files#r33733512>.
>